gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/step/map/AbstractAddElementStepPlaceholder.java

```diff
@@ -265,16 +260,12 @@ public void updateVariable(String name, Object value) {
 
     @Override
     public Collection<GValue<?>> getGValues() {
-        Set<GValue<?>> gValues = new HashSet<>();
+        Set<GValue<?>> gValues = GValueHelper.getGValuesFromProperties(properties);
```

Comment:
```suggestion
        Set<GValue<?>> gValues = new HashSet<>(GValueHelper.getGValuesFromProperties(properties));
```
There is a bit of risk to directly using the Set produced by GValueHelper.getGValuesFromProperties(properties) as it is currently generated by [Collectors.toSet() which is not guaranteed to be mutable](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collectors.html#toSet--)